aoservicelisting.civix.php

```diff
@@ -173,36 +193,142 @@ public static function endABARelationship($values, $rowNumber, $orgId) {
+      'id' => $cid,
+      $field => 'null',
+    ]);
+  }
+
+  public static function findDupes($cid, $orgId, &$individualParams, $rel = EMPLOYER_CONTACT_REL, $isOrgOptional = FALSE, $staffType) {
     if (!empty($cid)) {
-      $currentDetails = civicrm_api3('Contact', 'getsingle', ['id' => $cid]);
-      if ($currentDetails['first_name'] != $individualParams['first_name'] || $currentDetails['last_name'] != $individualParams['last_name']) {
-        $params = ['contact_id_a' => $cid, 'contact_id_b' => $orgId, 'is_active' => 1];
-        $relationships = civicrm_api3('Relationship', 'get', $params);
-        if (!empty($relationships['values'])) {
-          // End Date all relationships as they have either overwritten the data or not.
-          foreach ($relationships['values'] as $relationship) {
-            civicrm_api3('Relationship', 'create', ['id' => $relationship['id'], 'is_active' => 0, 'relationship_type_id' => $relationship['relationship_type_id'], 'end_date' => date('Y-m-d')]);
+      $currentDetails = civicrm_api3('Contact', 'getsingle', [
+        'id' => $cid,
+        'return' => ['first_name', 'last_name', REGULATED_URL, CERTIFICATE_NUMBER],
+      ]);
+      // Check, based on staff type
+      if ($staffType == 'regstaff') {
+        // Get saved regulator URL
+        if ($currentDetails['first_name'] != $individualParams['first_name'] &&
+          $currentDetails['last_name'] != $individualParams['last_name'] &&
+          $currentDetails[REGULATED_URL] != $individualParams['regulated_url']) {
+          // This is an overwrite on the record, check to see if this contact has an ABA certificate before ending the relationship.
+          if (!empty($currentDetails[CERTIFICATE_NUMBER])) {
+            // Move the certificate number elsewhere and delete the certificate number, don't end the relationship.
+            E::archiveField(CERTIFICATE_NUMBER, $currentDetails[CERTIFICATE_NUMBER], $cid);
+          }
+          else {
+            // There is no certificate number, we can now end the relationship.
+            E::terminateRelationship($cid, $orgId);
+          }
+        }
+        else {
+          // This is an edit, return the contact ID.
+          if ($currentDetails['first_name'] == $individualParams['first_name'] &&
+            $currentDetails['last_name'] == $individualParams['last_name'] &&
+            $currentDetails[REGULATED_URL] != $individualParams['regulated_url']) {
+            // This is an edit to only the regulated URL field, primary contacts can be edited.
+            $individualParams['contact_id'] = $cid;
+            return;
+          }
+          if (($currentDetails['first_name'] != $individualParams['first_name'] ||
+            $currentDetails['last_name'] != $individualParams['last_name']) &&
+            $currentDetails[REGULATED_URL] == $individualParams['regulated_url']) {
+            // This is an edit to the name, ensure that it is not a primary contact.
```

Comment:
Why do we need this check here? relationships would stay the same right?

Reply:
If we passed back the cid

Reply:
This is to prevent the primary contact from having it's name altered. The relationships would stay the same, but the name will be altered

Reply:
yeh ok makes sense maybe just expand the code comment to make that clear